Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -821,16 +821,27 @@ static void registerLogHook()
 
     static os_log_hook_t prevHook = nullptr;
 
-    prevHook = os_log_set_hook(OS_LOG_TYPE_DEFAULT, ^(os_log_type_t type, os_log_message_t msg) {
+#ifdef NDEBUG
+    // OS_LOG_TYPE_DEFAULT implies default, fault, and error.
+    constexpr auto minimumType = OS_LOG_TYPE_DEFAULT;
+#else
+    // OS_LOG_TYPE_DEBUG implies debug, info, default, fault, and error.
+    constexpr auto minimumType = OS_LOG_TYPE_DEBUG;
+#endif
+
+    prevHook = os_log_set_hook(minimumType, ^(os_log_type_t type, os_log_message_t msg) {
         if (prevHook)
             prevHook(type, msg);
 
         if (msg->buffer_sz > 1024)
             return;
 
-        // Skip debug logs in non-internal builds.
-        if (type == OS_LOG_TYPE_DEBUG)
+#ifdef NDEBUG
+        // Don't send messages with types we don't want to log in release. Even though OS_LOG_TYPE_DEFAULT is the minimum,
+        // the hook will be called for other subsystems with debug and info types.
+        if (type & (OS_LOG_TYPE_DEBUG | OS_LOG_TYPE_INFO))
```

Comment:
I think we want to send OS_LOG_TYPE_INFO?

Reply:
Yes, I think we only should return if type is OS_LOG_TYPE_DEBUG here.

Reply:
This only rejects `OS_LOG_TYPE_INFO` logs from other subsystems, with `minimumType = OS_LOG_TYPE_DEFAULT` the hook is not called for WebKit's subsystem for `OS_LOG_TYPE_INFO`.